### PR TITLE
Add support for arrays in the spatialite provider

### PR DESCRIPTION
**Pull request by pvalsecc**
New versions of GDAL/OGR (since trunk@35289) convert list types (StringList,
IntegerList, Integer64List and RealList) to a JSON string when it stores a
Spatialite table. It sets the column type as JSONSTRINGLIST, JSONINTEGERLIST,
JSONINTEGER64LIST or JSONREALLIST.

### CODE REVIEW

**Comment by nyalldawson** [comment]
@rouault you're probably best qualified to review this one, do you mind taking a look?

File: src/providers/spatialite/qgsspatialiteprovider.cpp [hunk 1]
```diff
@@ -591,6 +608,10 @@ QgsSpatiaLiteProvider::QgsSpatiaLiteProvider( QString const &uri )
   << QgsVectorDataProvider::NativeType( tr( "Text" ), "TEXT", QVariant::String )
   << QgsVectorDataProvider::NativeType( tr( "Decimal number (double)" ), "FLOAT", QVariant::Double )
   << QgsVectorDataProvider::NativeType( tr( "Whole number (integer)" ), "INTEGER", QVariant::LongLong )
+
+  << QgsVectorDataProvider::NativeType( tr( "Array of text (integer)" ), SPATIALITE_ARRAY_PREFIX.toUpper() + "TEXT" + SPATIALITE_ARRAY_SUFFIX.toUpper(), QVariant::StringList, 0, 0, 0, 0, QVariant::String )
```
Comment (rouault) [inline src/providers/spatialite/qgsspatialiteprovider.cpp]: The (integer) suffix is inappropriate here

File: src/providers/spatialite/qgsspatialiteprovider.cpp [hunk 2]
```diff
@@ -591,6 +608,10 @@ QgsSpatiaLiteProvider::QgsSpatiaLiteProvider( QString const &uri )
...
+  << QgsVectorDataProvider::NativeType( tr( "Array of decimal numbers (double)" ), SPATIALITE_ARRAY_PREFIX.toUpper() + "FLOAT" + SPATIALITE_ARRAY_SUFFIX.toUpper(), QVariant::List, 0, 0, 0, 0, QVariant::Double )
```
Comment (rouault) [inline src/providers/spatialite/qgsspatialiteprovider.cpp]: Should be REAL and not FLOAT to be interoperable with OGR

**Review by rouault — COMMENTED** [review]
Looks good to me, pending the correction for my comments

**Comment by pvalsecc** [comment]
@rouault, fixed your comments.

**Comment by rouault** [comment]
> @rouault, fixed your comments.

Did you push the related commit ?

**Comment by nyalldawson** [comment]
Glad to see I'm not the only one who thinks these changes are a step back... I wish they hadn't removed the public activity tab from users, that was a great way to track who was working on what in various qgis forks

**Pull request merged**